### PR TITLE
Add CpmTokenizerFast

### PR DESCRIPTION
**Pull request by JetRunner**
# What does this PR do?
Add a fast version of `CpmTokenizer`

Fixes #12837

### CODE REVIEW

File: src/transformers/models/cpm/tokenization_cpm_fast.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {"vocab_file": "spiece.model"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "TsinghuaAI/CPM-Generate": "https://huggingface.co/TsinghuaAI/CPM-Generate/resolve/main/spiece.model",
+    }
+}
```
Comment (sgugger) [inline src/transformers/models/cpm/tokenization_cpm_fast.py:28]:
```suggestion
VOCAB_FILES_NAMES = {"vocab_file": "spiece.model"}

PRETRAINED_VOCAB_FILES_MAP = {
    "vocab_file": {
        "TsinghuaAI/CPM-Generate": "https://huggingface.co/TsinghuaAI/CPM-Generate/resolve/main/spiece.model",
    }
}
```
Tokenizer fast also require the `tokenizer.json` file (as you can see in the tokenization fast file for XLNet) which should be generated if you locally create a `CpmTokenizerFast` and then use the `save_pretrained` method.

File: src/transformers/models/cpm/tokenization_cpm_fast.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            import jieba
+        except ModuleNotFoundError as error:
+            raise error.__class__(
+                "You need to install jieba to use CpmTokenizer or CpmTokenizerFast."
+                "See https://pypi.org/project/jieba/ for installation."
+            )
+        self.jieba = jieba
+        self.translator = str.maketrans(" \n", "\u2582\u2583")
+
+    def _tokenize(self, text, *args, **kwargs):
```
Comment (sgugger) [inline src/transformers/models/cpm/tokenization_cpm_fast.py:101]: The fast tokenizers do not use this method, so this code will never be reached. You will need to override `_encode_plus` and `_batch_encode_plus` in this case.

**Review by sgugger — COMMENTED** [review]
I don't think the fast tokenizer as it's written works for now, as the fast tokenizer do not call the `_tokenize` method.

**Comment by JetRunner** [comment]
> I don't think the fast tokenizer as it's written works for now, as the fast tokenizer do not call the `_tokenize` method.

Oops! It looks the old pull request isn't right. I'll take a closer look

**Comment by JetRunner** [comment]
@sgugger I've updated and tested it. It works fine!

**Review by sgugger — APPROVED** [review]
Looking good! I'll let you merge once the tokenizer file is uploaded!

**Comment by JetRunner** [comment]
Tokenizer file uploaded. Merging it.

**Pull request merged**